### PR TITLE
Web sdk build reland

### PR DESCRIPTION
**Pull request by eyebrowsoffire**
Trying to reland https://github.com/flutter/engine/pull/36520

A few things that are different:
* Addressed some comments from @zanderso 
* You can now use `felt build --host` to do the old host build, in case you want to run with `--local-engine` before we've made the changes to the flutter tool to support `--local-engine` from the wasm build.
* Changed how we do `--multi-root` in order to avoid accidentally pulling in any stuff from copied or built `dart-sdk` for the web sdk.

There were some issues with the google3 roll so I need to do a little more investigation to make sure I'm not breaking anything there before I land this.

### CODE REVIEW

File: common/config.gni [hunk 1]
```diff
@@ -22,6 +22,9 @@ declare_args() {
 
   # Whether to use a prebuilt Dart SDK instead of building one.
   flutter_prebuilt_dart_sdk = false
+
+  # Whether to build the flutter web sdk outline/DDC artifacts.
+  build_flutter_web_sdk = false
```
Comment (zanderso) [inline common/config.gni:27]: Following suit would make this `flutter_build_web_sdk`.

**Review by zanderso — APPROVED** [review]
gn lgtm w/ nit

File: lib/web_ui/dev/build.dart [hunk 2]
```diff
@@ -25,6 +25,12 @@ class BuildCommand extends Command<bool> with ArgUtils<bool> {
       'build-canvaskit',
       help: 'Build CanvasKit locally instead of getting it from CIPD. Disabled '
           'by default.',
+      defaultsTo: true
+    );
+    argParser.addFlag(
+      'host',
+      help: 'Build the host build instead of the wasm build, which is currently'
+          'needed for `flutter run --local-engine` to work'
```
Comment (mdebbar) [inline lib/web_ui/dev/build.dart:33]: ❤️

**Review by mdebbar — APPROVED** [review]
LGTM and thanks for adding the `--host` argument!

**Comment by a-siva** [comment]
Can this PR be reverted it appears to cause the Flutter HHH bot to fail, see https://logs.chromium.org/logs/dart/buildbucket/cr-buildbucket/8799750296394481121/+/u/gn_--full-dart-sdk_--no-prebuilt-dart-sdk/stdout

Also see   https://github.com/flutter/flutter/issues/113772 for the tracking issue.

**Comment by eyebrowsoffire** [comment]
We have merged a roll-forward fix: https://github.com/flutter/engine/pull/36926

**Pull request merged**